### PR TITLE
fix: bind aria attributes to ng-select in checkout (#14988)

### LINKED ISSUE #14988: AMP errors on Shipping Address and Payment Method forms
**Describe the bug**

- During checkout when Shipping Address and Payment Method forms are displayed while entering the information for the first time, it seems the forms have the same errors on the ng-select element.
- Same error is detected on the Address Book section of My Account.
- Screenshots have more details about the nature of the problem.

**Tell us the version of Spartacus**
- Library version: develop (5.0)

**Checkout**
Screen Shot 2022-02-09 at 5 38 04 PM
Screen Shot 2022-02-09 at 6 25 48 PM
Screen Shot 2022-02-09 at 6 25 29 PM
Screen Shot 2022-02-09 at 6 26 16 PM
Screen Shot 2022-02-09 at 6 26 07 PM

**Address Book**
Screen Shot 2022-02-09 at 6 23 26 PM

Screen Shot 2022-02-09 at 6 22 55 PM

Screen Shot 2022-02-09 at 6 22 17 PM

### PR DESCRIPTION
**Pull request by BenjaminEmiliani**
Closes #14988

### CODE REVIEW

File: projects/storefrontlib/shared/components/ng-select-a11y/ng-select-a11y.directive.ts [hunk 1]
```diff
@@ -22,15 +22,20 @@ export class NgSelectA11yDirective implements AfterViewInit {
...
-      divCombobox,
-      'aria-label',
-      this.cxNgSelectA11y.ariaLabel
-    );
-    this.renderer.setAttribute(
-      divCombobox,
-      'aria-controls',
-      this.cxNgSelectA11y.ariaControls
-    );
+    if (this.cxNgSelectA11y.ariaLabel) {
```
Comment (Pucek9) [inline projects/storefrontlib/shared/components/ng-select-a11y/ng-select-a11y.directive.ts]: why check it if it's a required field?

File: projects/storefrontlib/shared/components/ng-select-a11y/ng-select-a11y.directive.ts [hunk 2]
```diff
@@ -22,15 +22,20 @@ export class NgSelectA11yDirective implements AfterViewInit {
...
+      this.renderer.setAttribute(
+        divCombobox,
+        'aria-label',
+        this.cxNgSelectA11y.ariaLabel
+      );
+    }
+
+    if (this.cxNgSelectA11y.ariaControls) {
```
Comment (Pucek9) [inline projects/storefrontlib/shared/components/ng-select-a11y/ng-select-a11y.directive.ts]: the same here, but I can see we have so many cases where `aria-controls` is the same as in element, so maybe we can do it optional and use `id` of  `this.elementRef`

**Comment by FollowTheFlo** [comment]
Amp tests shows no errors for Checkout Delivery Address and Payment Forms
image

File: projects/storefrontlib/shared/components/ng-select-a11y/ng-select-a11y.directive.ts (hunk 1 above)
Comment (BenjaminEmiliani) [reply projects/storefrontlib/shared/components/ng-select-a11y/ng-select-a11y.directive.ts]: Well the object literal may be passed as an input but it is possible to not pass the property ariaLabel inside the object

File: projects/storefrontlib/shared/components/ng-select-a11y/ng-select-a11y.directive.ts (hunk 2 above)
Comment (FollowTheFlo) [reply projects/storefrontlib/shared/components/ng-select-a11y/ng-select-a11y.directive.ts]: I find the code more readable when we see the 2 parameters passed like it is at the moment, even though DRY is not strictly applied with element id copied.

File: projects/storefrontlib/shared/components/ng-select-a11y/ng-select-a11y.directive.ts (hunk 1 above)
Comment (BenjaminEmiliani) [reply projects/storefrontlib/shared/components/ng-select-a11y/ng-select-a11y.directive.ts]: @Pucek9 so the field is not strictly required

**Comment by FollowTheFlo** [comment]
Tested with the 3 SRs, there is an issue with NVDA (Chrome and Firefox), 
when selecting a value (with enter or mouse), 
**the selection value is not read. It just says 'edit required invalid entry has auto complete blank'.** 
Occurs on all ng-select (Titles and countries selection)
image

In comparison with 'develop' branch:
'Title Mr. edit has auto complete blank'
image
Difference in Accessibility tree:
An extra generic node
image

File: projects/storefrontlib/shared/components/ng-select-a11y/ng-select-a11y.directive.ts [hunk 3]
```diff
@@ -22,15 +23,16 @@ export class NgSelectA11yDirective implements AfterViewInit {
...
+    const elementId = this.elementRef.nativeElement.id;
+    const _ariaControls = this.cxNgSelectA11y.ariaControls ?? elementId;
+
+    if (_ariaLabel) {
+      this.renderer.setAttribute(divCombobox, 'aria-label', _ariaLabel);
+    }
+
+    if (_ariaControls) {
+      this.renderer.setAttribute(divCombobox, 'aria-controls', _ariaControls);
+    }
```
Comment (Pucek9) [inline projects/storefrontlib/shared/components/ng-select-a11y/ng-select-a11y.directive.ts]:
```suggestion
    const ariaLabel = this.cxNgSelectA11y.ariaLabel;
    const elementId = this.elementRef.nativeElement.id;
    const ariaControls = this.cxNgSelectA11y.ariaControls ?? elementId;

    this.renderer.setAttribute(divCombobox, 'aria-label', ariaLabel);

    if (ariaControls) {
      this.renderer.setAttribute(divCombobox, 'aria-controls', ariaControls);
    }
```

File: projects/storefrontlib/shared/components/ng-select-a11y/ng-select-a11y.directive.ts [hunk 4]
```diff
@@ -12,7 +12,8 @@ import {
 export class NgSelectA11yDirective implements AfterViewInit {
   /**
    * Use directive to bind aria attribute to inner element of ng-select
-   * Angular component for accessibility compliance.
+   * Angular component for accessibility compliance. If ng-select controls itself
+   * ariaControls is not needed, instead bind a specific id to the <ng-select> element.
    */
   @Input() cxNgSelectA11y: { ariaLabel: string; ariaControls: string };
```
Comment (Pucek9) [inline projects/storefrontlib/shared/components/ng-select-a11y/ng-select-a11y.directive.ts]:
```suggestion
  @Input() cxNgSelectA11y: { ariaLabel: string; ariaControls?: string };
```

**Review by developpeurweb — APPROVED** [review]
Works well with JAWS with no AMP violations, fulfills audit requirements. Please merge as soon as the code is approved by an angular dev.
Let's create a new ticket to deal with the NVDA bug separately.

**Comment by developpeurweb** [comment]
> Tested with the 3 SRs, there is an issue with NVDA (Chrome and Firefox), when selecting a value (with enter or mouse), **the selection value is not read. It just says 'edit required invalid entry has auto complete blank'.** Occurs on all ng-select (Titles and countries selection) image
> 
> In comparison with 'develop' branch: 'Title Mr. edit has auto complete blank' image Difference in Accessibility tree: An extra generic node image

@FollowTheFlo just so we don't lose track of this, can you please transfer your comments and screenshots to a new ticket for the backlog column, labels: `a11y/cross-sr`, `bug`, `minor`.

**Review by FollowTheFlo — APPROVED** [review]
i created minor bug for NVDA #15422 
Code change suggested by Michal are applied.

**Pull request merged**